src/main/java/homework_2/random_chars_table/RandomCharsTable.java

```diff
@@ -0,0 +1,78 @@
+package homework_2.random_chars_table;
+
+import java.util.InputMismatchException;
+import java.util.Random;
+import java.util.Scanner;
+
+public class RandomCharsTable {
+    private static int length;
+    private static int width;
+
+    public static void run() {
+        System.out.println("Enter valid table length, width and strategy(even or odd) in format \"length width strategy\" ");
+        getParameters();
+    }
+
+    private static void getParameters() {
+        try (Scanner scanner = new Scanner(System.in)) {
+            length = scanner.nextInt();
+            width = scanner.nextInt();
+            String strategy = scanner.next();
+            checkLengthAndWidth(length, width);
+            checkStrategyValid(strategy);
+            char[][] table = new char[length][width];
+            fillAndPrintTable(strategy, table);
+        } catch (NegativeArraySizeException e) {
+            System.out.println("Array size cannot be negative.");
+        } catch (InputMismatchException e) {
+            System.out.println("Input parameters must be in the format [positive integer(>0)] [positive integer(>0)] [even|odd]");
+        }
+    }
+
+    private static void checkLengthAndWidth(int length, int width) {
+        if (length == 0 || width == 0){
```

Comment:
```suggestion
        if (length <= 1 || width <= 1) {
```
covers several cases